daml-assistant/src/DAML/Assistant/Env.hs

```diff
@@ -57,6 +59,15 @@ testDamlEnv Env{..} = firstJustM (\(test, msg) -> unlessMaybeM test (pure msg))
       , "The project directory does not exist. Please check if DAML_PROJECT is incorrectly set.")
     ]
 
+-- | Determine the absolute path to the assistant.
+getDamlAssistantPath :: Applicative f => DamlPath -> f DamlAssistantPath
+getDamlAssistantPath (DamlPath damlPath)
```

Comment:
Vomit. Why does our packaging logic for Haskell do that? Can we fix it at source?

Reply:
Our packaging logic does this because producing reasonably portable, dynamically linked binaries on Linux is fairly tricky. My attempts at fixing it so far sadly weren’t particularly successful and so far I wanted to avoid having to patch our dynamic linker :smile:

Reply:
Fair enough. Could we write an HLint rule banning getExecutablePath, if it's never going to be correct? I imagine whatever you learnt was hard won knowledge I don't want to get lost.

Reply:
`getExecutablePath` is still usable, and we do use it below. It just doesn't return what you might expect based on the docs . . .